### PR TITLE
Install removes packages

### LINKED ISSUE #61: Package installation deletes necessary modules
`mosaic` library npm installation removes `react`, `react-dom`, ... from `node_modules` directory.

Steps to reproduce the issue: 
1. npx create-react-app my-app --template typescript
2. npm start *# test*
The application works.

3. npm i @melfore/mosaic ***# removes react etc***
4. npm start *# test*
An error is raised: `Module not found: Can't resolve 'react' in '/home/mas/dev/tmp/prova_import/my-app/src'`

### LINKED ISSUE #62: Wrong import path of npm library
To import mosaic from an external application, `dist` must be appended.

Example:

```js
import { Button } from '@melfore/mosaic';
```
doesn't work.

```js
import { Button } from '@melfore/mosaic/dist';
```
works

### PR DESCRIPTION
**Pull request by luciobordonaro**
This should fix #61 and fixes #62 .
Please read the notes below carefully as they are crucial to understand the project structure from now on. But if you want to skip read the TLDR at least 😅 
I've tested it only locally, it will for sure require a CI run on our final user project.

**TLDR**: We don't have `dependencies` any longer as we rely on `devDependencies` for developers and `peerDependencies` for final users, forcing final users to have `react`, `material-ui` and `react-intl` in their own `package.json`.

---

**#62 - Wrong import path of npm library**
It got fixed with [this simple commit](https://github.com/melfore/mosaic/pull/67/commits/eefd278b0fc5d27080b784729be3dca96ceca881). I didn't update the reference for declarations of types since the beginning of the project, now it is correct.

---

**#61 - Package installation deletes necessary modules**
It was complicated because project dependencies weren't setup as they should. It required a lot of work and some radical changes to **Mosaic** structure. 

Since our components library is built on top of `react`, `material-ui` and `react-intl` we can leverage existing install of these packages in the final project that will import us. This can be done moving all existing `dependencies` to `devDependencies` (life of contributors won't change), and listing all the previous `dependencies` inside `peerDependencies`.
This is the suggested way to create `npm` packages build on top of other libraries.

What happens is that when someone installs `@melfore/mosaic` the dependencies listed into `peerDependencies` won't be installed. Our library will use the ones available in the project. If some of them are missing `npm install @melfore/mosaic` will show a warning to the user to install them manually, otherwise some features might not work.

Moving `react-*`, `material-ui` and `react-intl` to the right place was easy, what wasn't easy was moving `react-scripts` (the culprit of log posted by @erosval ).
The reason is that `react-scripts` was used both by underlying commands and from scripts contained in our `package.json` (`storybook` and `test` ones). So, in addition to work done before I also removed `react-scripts` totally from our project, replacing it with standalone commands. 
For `storybook` we got no drawbacks, while for `jest` we got a minor one, coverage is computed differently so we might need to adjust test to make it count (I've reduced it temporarily to 75%, will create a dedicated issue for that and address it later on).

These are our requirements to work (also shown in README.md):
``` 
"peerDependencies": {
    "@material-ui/core": "4.9.x",
    "@material-ui/icons": "4.9.x",
    "react": "16.13.x",
    "react-dom": "16.13.x",
    "react-intl": "4.3.x"
},
```

### CODE REVIEW

**Comment by masmau** [comment]
The error documented in #70 is still present.
